### ISSUE TITLE
CA1056 should ignore JSON properties

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn-analyzers#5133]
### Analyzer

**Diagnostic ID**: [CA1056](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1056): `URI properties should not be strings`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.3 (Latest)

### Describe the bug

```csharp
    public sealed class Redirection
    {
        [JsonPropertyName("source_path")]
        public string SourcePath { get; set; } = null!;

        [JsonPropertyName("redirect_url")]
        public string RedirectUrl { get; set; } = null!;
    }
```

This possibly shouldn't trigger a diagnostic.

**Comment by jmarolf** [comment]
Json serializer can deserialize an Uri. Would need an explanation of why the heuristic should ignore these.

**Comment by Youssef1313** [comment]
@jmarolf I haven't tried, but I don't see it mentioned in docs [here](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-supported-collection-types?pivots=dotnet-5-0).

**Comment by jmarolf** [comment]
Please file a bug on the runtime team to document this. Looking at the runtime tests [here](https://github.com/dotnet/runtime/blob/e98d043d7d293c88a346b632d8fc12564a8ef0ce/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/Value.ReadTests.cs#L38) I assume that Uris are fully supported
